### PR TITLE
Fix IPLB configuration page

### PR DESCRIPTION
**Pull request by FredericEspiau**
## Fix IPLB configuration page
MANAGER-1930
### Description of the Change

- replace `oui-table` by `oui-datagrid`
- `oui-progress` displays progression text
-  fix crash when sending data to API
- polling works correctly

### CODE REVIEW

**Comment by miton18** [comment]
Fix #1164

File: client/app/iplb/configuration/iplb-configuration.controller.js [hunk 1]
```diff
@@ -27,26 +27,25 @@ class IpLoadBalancerConfigurationCtrl {
...
+  applyChanges(targets) {
     let promise = this.$q.resolve([]);
-    if (zone) {
+
+    if (!_.isArray(targets)) {
       promise = this.IpLoadBalancerConfigurationService
-        .refresh(this.$stateParams.serviceName, zone);
-    } else if (this.selectedZones.length === this.zones.length) {
+        .refresh(this.$stateParams.serviceName, targets);
+    }
```
Comment (marie-j) [inline client/app/iplb/configuration/iplb-configuration.controller.js]:
```suggestion
    const promise = !_.isArray(targets) ? this.IpLoadBalancerConfigurationService
        .refresh(this.$stateParams.serviceName, targets) : this.$q.resolve([])
```

File: client/app/iplb/configuration/iplb-configuration.html [hunk 2]
```diff
@@ -2,29 +2,64 @@
...
+        <extra-top>
+            <oui-button
+                class="mb-3"
+                data-variant="secondary"
+                data-text="{{:: 'iplb_configuration_action_apply' | translate }}"
+                data-on-click="$ctrl.applyChanges($selectedRows)"
+                data-disabled="!$selectedRows.length"></oui-button>
+        </extra-top>
+        <oui-column
+            data-title="'iplb_configuration_datacenter' | translate"
```
Comment (marie-j) [inline client/app/iplb/configuration/iplb-configuration.html]:
```suggestion
            data-title=":: 'iplb_configuration_datacenter' | translate"
```

File: client/app/iplb/configuration/iplb-configuration.html [hunk 3]
```diff
@@ -2,29 +2,64 @@
...
+            data-property="name"
+            data-sortable="asc"></oui-column>
+        <oui-column
+            data-title="'iplb_configuration_progress' | translate"
+            data-sortable>
+            <oui-progress
+                data-compact
+                data-ng-if="$row.task">
+                <oui-progress-bar
+                    data-type="{{ $ctrl.ServiceHelper.constructor.getTaskProgressType($row.task.status) }}"
```
Comment (marie-j) [inline client/app/iplb/configuration/iplb-configuration.html]: No `::`

Comment (FredericEspiau) [reply client/app/iplb/configuration/iplb-configuration.html]: Well it's a progress bar, I do hope the status changes with the progression :)

File: client/app/cloud/project/compute/infrastructure/privateNetwork/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "cpci_private_network_list_header_disabled_notice": "Désactivé",
   "cpci_private_network_list_vrack_unnamed": "Sans nom",
   "cpci_private_network_list_untagged_vlan": "Aucun VLAN",
-  "cpci_private_network_activateVrack_description": "Une fois votre projet connecté au vRack, reliez vos Serveurs Dédiés, instances Public Cloud et Private Cloud, entre tous les datacentres OVH. Vous pourrez créer près de 4000 vLANs!",
+  "cpci_private_network_activateVrack_description": "Une fois votre projet connecté au vRack, reliez vos serveurs dédiés, instances Public Cloud et Private Cloud, entre tous les datacenters OVH. Vous pourrez créer près de 4000 vLANs!",
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/infrastructure/privateNetwork/translations/Messages_fr_FR.json]: All text changes are validated by Content

File: client/app/cloud/project/compute/translations/Messages_fr_FR.json [hunk 5]
```diff
@@ -6,12 +6,12 @@
   "cpc_tab_loadbalancer": "Load Balancer",
   "cpc_tab_project": "Gestion et conso. du projet",
   "cpb_tab_quota": "Quota",
-  "cpc_message": "Le Cloud OVH arrive en Allemagne ! <a href=\"http://superlienverslapage\">Découvrez notre datacentre de Limburg.</a>",
+  "cpc_message": "Le Cloud OVH arrive en Allemagne ! <a href=\"http://superlienverslapage\">Découvrez notre datacenter de Limburg.</a>",
```
Comment (marie-j) [inline client/app/cloud/project/compute/translations/Messages_fr_FR.json]: Anchor could be extracted and I'm not sure that link is valid 😄

File: client/app/helper/service-helper.service.js [hunk 6]
```diff
@@ -8,37 +8,48 @@
...
+          ? pathsToGetErrorMessage
+          : [pathsToGetErrorMessage];
+        const standardPaths = [
+          'data.message',
+          'message',
+          'data'];
+
+        const paths = [...firstPaths, ...standardPaths];
+
+        return (_.isArray(err) ? err : [err]).map((currentError) => {
```
Comment (JeremyDec) [inline client/app/helper/service-helper.service.js]: `error` ? :)

File: client/app/iplb/configuration/iplb-configuration.controller.js [hunk 7]
```diff
@@ -22,46 +41,77 @@ class IpLoadBalancerConfigurationCtrl {
...
-    this.selectedZones = [];
-  }
+    const zoneData = _.has(this.zones, 'data') ? this.zones.data : this.zones;
 
-  onSelectionChange(selection) {
-    this.selectedZones = selection;
-  }
+    const targets = _.isArray(zone)
+      ? zone
+      : [this.zones.data.find(currentZone => currentZone.id === zone)];
```
Comment (JeremyDec) [inline client/app/iplb/configuration/iplb-configuration.controller.js]: You can destructure : 
```suggestion
      : [this.zones.data.find(({ id }) => id === zone)];
```

File: client/app/iplb/configuration/iplb-configuration.controller.js [hunk 8]
```diff
@@ -22,46 +41,77 @@ class IpLoadBalancerConfigurationCtrl {
...
 
-    promise.then(() => {
-      this.startPolling();
-      if (this.poller) {
-        this.poller.$promise.then(() => {
+    promise
+      .then(() => {
+        this.zones.data
+          .filter(currentZone => targetsToApplyChangesTo
+            .find(target => target.name === currentZone.name))
```
Comment (JeremyDec) [inline client/app/iplb/configuration/iplb-configuration.controller.js]:
```suggestion
            .find(({ name }) => name === currentZone.name))
```

File: client/app/iplb/configuration/iplb-configuration.html (hunk 3 above)
Comment (JeremyDec) [reply client/app/iplb/configuration/iplb-configuration.html]: No, in fact the progress bar type property has one-time binding, so `::` is possible I guess.
See https://github.com/ovh-ux/ovh-ui-angular/blob/master/packages/oui-progress/src/bar/progress-bar.controller.js#L13.

File: client/app/iplb/configuration/iplb-configuration.html [hunk 9]
```diff
@@ -1,30 +1,65 @@
...
+                data-translate="iplb_configuration_changes_0"></span>
+            <span
+                data-ng-if="$row.changes === 1"
+                class="oui-status oui-status_warning"
+                data-translate="iplb_configuration_changes_1"></span>
+            <span
+                data-ng-if="$row.changes > 1"
+                class="oui-status oui-status_warning"
+                data-translate="iplb_configuration_changes_count"
+                data-translate-values="{ count: $row.changes }"></span>
```
Comment (JeremyDec) [inline client/app/iplb/configuration/iplb-configuration.html]: these three span cannot be done with only one and ngPluralize ?

File: client/app/helper/service-helper.service.js (hunk 6 above)
Comment (FredericEspiau) [reply client/app/helper/service-helper.service.js]: Would `erro` make you happy ?

File: client/app/iplb/configuration/iplb-configuration.html (hunk 9 above)
Comment (FredericEspiau) [reply client/app/iplb/configuration/iplb-configuration.html]: Did try this and wasn't happy with the result !

Comment (FredericEspiau) [reply client/app/iplb/configuration/iplb-configuration.html]: Done, it works great !

File: client/app/helper/service-helper.service.js (hunk 6 above)
Comment (FredericEspiau) [reply client/app/helper/service-helper.service.js]: Done

File: client/app/iplb/configuration/iplb-configuration.controller.js (hunk 8 above)
Comment (FredericEspiau) [reply client/app/iplb/configuration/iplb-configuration.controller.js]: Done

File: client/app/iplb/configuration/iplb-configuration.controller.js (hunk 7 above)
Comment (FredericEspiau) [reply client/app/iplb/configuration/iplb-configuration.controller.js]: Done

File: client/app/helper/service-helper.service.js [hunk 10]
```diff
@@ -68,19 +79,20 @@
...
 
     successHandler(message, containerName) {
-      return (data) => {
+      return data => (_.isArray(data) ? data : [data]).map((datum) => {
+        let messageToWrite = this.$translate.instant(defaultSuccessMessage);
+
         if (message) {
-          const jsonData = data ? data.toJSON ? data.toJSON() : data : {}; // eslint-disable-line
-          this.CloudMessage.success(_.isString(message)
+          const jsonData = _.isFunction(datum, 'toJSON') ? datum.toJSON() : datum || {};
```
Comment (cbourgois) [inline client/app/helper/service-helper.service.js]: -- `eslint-disable-line`
Nice !

**Comment by jleveugle** [comment]
Hello @FredericEspiau,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

File: client/app/iplb/configuration/iplb-configuration.html [hunk 11]
```diff
@@ -1,30 +1,65 @@
...
-                </oui-progress-bar>
+    <oui-datagrid
+        data-ng-if="!$ctrl.zones.loading"
+        data-rows="$ctrl.zones.data"
+        data-selectable-rows>
+        <extra-top>
+            <oui-button
+                class="mb-3"
+                data-variant="secondary"
+                data-text="{{:: 'iplb_configuration_action_apply' | translate }}"
```
Comment (antleblanc) [inline client/app/iplb/configuration/iplb-configuration.html]: FYI the `data-text` attribut is deprecated.

File: client/app/iplb/configuration/iplb-configuration.html [hunk 12]
```diff
@@ -1,30 +1,65 @@
...
+                    '0': ('iplb_configuration_changes_0' | translate),
+                    '1': ('iplb_configuration_changes_1' | translate),
+                    'other': ('iplb_configuration_changes_count' | translate: { count: $row.changes }),
+                }"></span>
+        </oui-column>
+        <oui-action-menu
+            data-align="end"
+            data-compact>
+            <oui-action-menu-item
+                data-text="{{:: 'iplb_configuration_action_apply' | translate}}"
```
Comment (antleblanc) [inline client/app/iplb/configuration/iplb-configuration.html]: FYI the `data-text` attribut is deprecated.

**Pull request merged**